gen-frcnetcomm.py

```diff
@@ -0,0 +1,249 @@
+                # Read until an enum is encountered
+                line = ""
+                pos = -1
+                while line.find("enum") == -1:
+                    line = cppSource.readline()
+                    if line == "":
+                        break
+
+                if line == "":
+                    break
+
+                # If "{" is on next line, read next line
+                if line.find("{") == -1:
+                    line = cppSource.readline()
+
+                # Write enum to output file as interface
+                values = []
+                line = cppSource.readline()
+                while line.find("}") == -1:
+                    if line == "\n":
+                        values.append("")
+                    else:
+                        if line[0] != "#":
+                            line = line.strip()
+                            if line[len(line) - 1] == ",":
+                                values.append(line[0:len(line) - 1])
+                            else:
+                                values.append(line)
+                    line = cppSource.readline()
+
+                # Extract enum name
+                nameStart = 0
+                for i, c in enumerate(line):
+                    if c != " " and c != "}":
+                        nameStart = i
+                        break
+                enumName = line[nameStart:len(line) - 2]
+
+                # Write interface name
+                temp.write("  /**\n"
+                           "   * ")
+
+                enumCamel = re.findall(r'[A-Z](?:[a-z]+|[A-Z]*(?=[A-Z]|$))', enumName)
+                temp.write(enumCamel[0] + " ")
+                for i in range(1, len(enumCamel)):
+                    temp.write(enumCamel[i][0].lower() + enumCamel[i][1:len(enumCamel[i])] + " ")
+                temp.write("from " + fileName + "\n"
+                           "   */\n"
+                           "  @SuppressWarnings(\"TypeName\")\n"
+                           "  public interface " + enumName + " {\n")
+
+                # Write enum values
+                count = 0
+                for value in values:
+                    # Pass through empty lines
+                    if value == "":
+                        temp.write("\n")
+                        continue
+
+                    if value.find("=") == -1:
+                        value = value + " = " + str(count)
+                        count += 1
+
+                    # Add scoping for values from a different enum
+                    if enumName != "tModuleType" and value.find("kModuleType") != -1:
+                        value = value.replace("kModuleType", "FRCNetworkCommunicationsLibrary.tModuleType.kModuleType")
+                        value = value.replace("= ", "=\n        ")
+                    temp.write("    int " + value[0:len(value)] + ";\n")
+
+                temp.write("  }\n\n")
+
+    # Write rest of generated file
+    temp.write("""\
```

Comment:
Doesn't seem to. I made the same change in license-update.py and it didn't change the output.